src/main/resources/files/liberty/TestSecureController.java.tpl

```diff
@@ -75,28 +67,26 @@ public class TestSecureController {
 
         token.setGroups(Arrays.asList("user", "protected"));
 
-        JWSObject jwsObject = new JWSObject(header, new Payload(token.toJSONString()));
-
-        // Apply the Signing protection
-        JWSSigner signer = new RSASSASigner(key);
+        return provider.generateToken(new JsonObject().mergeIn(token.toJSONString()), new JWTOptions().setAlgorithm("RS256"));
```

Comment:
Probably something from VertX JWT library, but why do you need to define RS256 here again? We did it already when creating the 'provider'.

Reply:
@rdebusscher It has to be like that. It is probably Vertx JWT UX thing. If I remove it, I get 
```
java.lang.RuntimeException: Algorithm not supported
    at io.vertx.ext.jwt.JWT.sign(JWT.java:347)
    at io.vertx.ext.auth.jwt.impl.JWTAuthProviderImpl.generateToken(JWTAuthProviderImpl.java:166)
```

So the current code as committed it correct.